concordium-std/src/test_infrastructure.rs

```diff
@@ -695,6 +655,313 @@ impl<T: AsRef<[u8]>> Seek for ContractStateTest<T> {
     }
 }
 
+impl HasCallResponse for Cursor<Vec<u8>> {
+    fn size(&self) -> u32 { self.data.len() as u32 }
+}
+
+/// Holds a function used for mocking invocations of contracts with
+/// `invoke_contract`.
+/// The provided closure may mutate its environment, for example to keep track
+/// of how many times the function was called during testing. This is the reason
+/// for `FnMut` bound.
```

Comment:
This conflicts with the docs for `TestMockFn`.

Reply:
Should we make it `Fn` for now? Or what are the awkward patterns for mutating the environment that you mention?

Reply:
Let's keep it fnmut. If I missed updating documentation please update it.